core/mutex.c

```diff
@@ -54,12 +55,11 @@ void mutex_lock(struct mutex_t *mutex)
 static void mutex_wait(struct mutex_t *mutex)
 {
     unsigned irqstate = disableIRQ();
-    DEBUG("%s: Mutex in use. %u\n", sched_active_thread->name, mutex->val);
+    DEBUG("%s: Mutex in use. %u\n", sched_active_thread->name, ATOMIC_VALUE(mutex->val));
 
-    if (mutex->val == 0) {
+    if (atomic_set_to_one(&mutex->val)) {
```

Comment:
In the current version the value remains unchanged - in your version it might be changed.